src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/EncoderManager.m

```diff
@@ -355,7 +366,8 @@ - (void)setContext:(MTLContex * _Nonnull)mtlc {
   //
   jboolean needEnd = JNI_FALSE;
   if (_encoder != nil) {
-    if (_destination != dest || renderOptions->isAA != _encoderStates.aa) {
+    if (_destination != dest || renderOptions->isAA != _encoderStates.aa ||
+            renderOptions->isAAShader != _encoderStates.aaShader) {
```

Comment:
Hi Alexey,

Is there any performance impact because of ending the encoder on switch?

Thanks,
Jay

Reply:
Hi Jay,

Thanks for the comment. I didn’t find any difference at least with RenderPerfTest. However, looks like we don’t need to end the encoder here - the appropriate state will be recovered later in the code (by updatePipelineState method).

So, I’m going to remove this condition from the if statement.

Best Regards,
Alexey

Reply:
Hi Alexey,

Yes i also think we dont need to explicit end encoder here and leave it to updatePipelineState. This was the next clarification i wanted to ask.

Thanks,
Jay